Infrastructure/Repositories/GenericRepository.cs

```diff
@@ -75,6 +73,21 @@ public virtual async Task<T> FindByIdAsync(Guid id, params string[] navigationPr
         return entity;
     }
 
+    public virtual async Task<T> FoundOrThrowAsync(Guid id, string message, params string[] navigationProperties)
+    {
+        var query = ApplyNavigation(navigationProperties);
+        T entity = await query.FirstOrDefaultAsync(e => e.Id.Equals(id));
+        if (entity is null)
+        {
+            if (string.IsNullOrEmpty(message))
```

Comment:
use [Optional Argument](https://learn.microsoft.com/en-us/dotnet/csharp/programming-guide/classes-and-structs/named-and-optional-arguments?redirectedfrom=MSDN) like this